x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -6,64 +6,118 @@
-  defaultTitle,
   bounds,
   initialInput,
-  job,
   onCreate,
   onCancel,
+  mlApiServices,
 }) => {
-  const isNewJob = initialInput?.jobIds !== undefined && initialInput?.jobIds[0] !== job.job_id;
+  const isMounted = useMountedState();
+  const newJobUrl = useMlLink({ page: ML_PAGES.ANOMALY_DETECTION_CREATE_JOB });
+  const [jobIds, setJobIds] = useState(initialInput?.jobIds ?? []);
 
-  const [panelTitle, setPanelTitle] = useState<string>(defaultTitle);
+  const initialValuesRef = useRef({
+    titleManuallyChanged: !!initialInput?.title,
+    isNewJob:
+      initialInput?.jobIds !== undefined && jobIds.length && initialInput?.jobIds[0] !== jobIds[0],
+  });
+
+  const [job, setJob] = useState<MlJob | undefined>();
+  const [panelTitle, setPanelTitle] = useState<string>(initialInput?.title ?? '');
   const [functionDescription, setFunctionDescription] = useState<string | undefined>(
     initialInput?.functionDescription
   );
   // Reset detector index and entities if the job has changed
   const [selectedDetectorIndex, setSelectedDetectorIndex] = useState<number>(
-    !isNewJob && initialInput?.selectedDetectorIndex ? initialInput.selectedDetectorIndex : 0
+    !initialValuesRef.current.isNewJob && initialInput?.selectedDetectorIndex
+      ? initialInput.selectedDetectorIndex
+      : 0
   );
   const [selectedEntities, setSelectedEntities] = useState<MlEntity | undefined>(
-    !isNewJob && initialInput?.selectedEntities ? initialInput.selectedEntities : undefined
+    !initialValuesRef.current.isNewJob && initialInput?.selectedEntities
+      ? initialInput.selectedEntities
+      : undefined
   );
-
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
   const isPanelTitleValid = panelTitle.length > 0;
 
+  useEffect(
+    function setUpPanel() {
+      if (isMounted()) {
+        async function fetchJob() {
+          const { jobs } = await mlApiServices.getJobs({ jobId: jobIds.join(',') });
+
+          if (jobs.length > 0) {
+            // Reset values if the job has changed
+            if (initialValuesRef.current.isNewJob) {
+              setSelectedDetectorIndex(0);
+              setSelectedEntities(undefined);
+              setFunctionDescription(undefined);
+            }
+
+            setJob(jobs[0]);
+            setErrorMessage(undefined);
+          }
+        }
+
+        if (jobIds.length === 1) {
+          if (!initialValuesRef.current.titleManuallyChanged) {
+            setPanelTitle(getDefaultSingleMetricViewerPanelTitle(jobIds[0]));
+          }
+          if (mlApiServices && jobIds.length === 1 && jobIds[0] !== job?.job_id) {
+            fetchJob().catch((error) => {
+              const errorMsg = extractErrorMessage(error);
+              setErrorMessage(errorMsg);
+            });
+          }
+        }
+      }
+    },
+    [isMounted, jobIds, mlApiServices, panelTitle, initialValuesRef.current.isNewJob, job?.job_id]
```

Comment:
Just realized, both here and in the anomaly swim lane initializer, we need to introduce a debounce on job IDs change.

Reply:
🤔 I'm not sure we do in this case since we only do the fetch if a new job has been selected (that hasn't already been loaded). Also - are people rapidly clicking around a ton on job ids? I think the dropdown closes after each selection (at least on single selection) so it would be difficult for a user to be changing job ids extremely quickly.